test/sdn_migration/sdn_migration_test.go

```diff
@@ -0,0 +1,525 @@
+		}
+	}
+
+	return criticalAlertCount, warningAlertCount, nil
+}
+
+// getEnvVar returns the env variable value and if unset returns default provided
+func getEnvVar(key, value string) string {
+	result, exist := os.LookupEnv(key)
+	if exist {
+		return result
+	}
+	return value
+}
+
+// osdClusterReadyHealthCheck verifies the osd-cluster-ready health check job is passing
+func osdClusterReadyHealthCheck(ctx context.Context, clusterClient *openshiftclient.Client, action, reportDir string) error {
+	var (
+		err error
+		job batchv1.Job
+	)
+
+	if err = clusterClient.Get(ctx, osdClusterReadyJobName, "openshift-monitoring", &job); err != nil {
+		return fmt.Errorf("failed to get existing %s job %v", osdClusterReadyJobName, err)
+	}
+
+	newJob := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: osdClusterReadyJobName,
+			Annotations:  job.Annotations,
+			Namespace:    job.Namespace,
+		},
+		Spec: job.Spec,
+	}
+
+	newJob.Spec.Selector.MatchLabels = map[string]string{}
+	newJob.Spec.Template.ObjectMeta.Name = newJob.GetGenerateName()
+	newJob.Spec.Template.ObjectMeta.Labels = map[string]string{}
+	newJob.Spec.Template.Spec.Containers[0].Name = newJob.GetGenerateName()
+
+	if err = clusterClient.Create(ctx, newJob); err != nil {
+		return fmt.Errorf("failed to create %s job: %v", newJob.GetName(), err)
+	}
+
+	defer func() {
+		_ = clusterClient.Delete(ctx, newJob)
+	}()
+
+	return clusterClient.OSDClusterHealthy(ctx, newJob.GetName(), reportDir, osdClusterReadyJobTimeout)
+}
+
+type upgradeError struct {
+	err error
+}
+
+// Error returns the formatted error message when upgradeError is invoked
+func (e *upgradeError) Error() string {
+	return fmt.Sprintf("osd upgrade failed: %v", e.err)
+}
+
+func getOpenshiftConfig(ctx context.Context, dynamicClient *dynamic.DynamicClient, resource string, name string) (*unstructured.Unstructured, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: resource,
+	}
+	return dynamicClient.Resource(gvr).Get(ctx, name, metav1.GetOptions{})
+}
+
+// checkMigrationStatus probes the status of the SDN-to-OVN migration
+func checkMigrationStatus(ctx context.Context, client *openshiftclient.Client) error {
+	var (
+		err     error
+		network configv1.Network
+	)
+
+	for i := 1; i <= upgradeMaxAttempts; i++ {
+		// Get the current network configuration
+		err = client.Get(ctx, "cluster", "", &network)
+		if err != nil {
+			return fmt.Errorf("failed to get network configuration: %v", err)
+		}
+
+		// Check for the migration condition
+		migrationInProgress := false
+		for _, cond := range network.Status.Conditions {
+			if cond.Type == "NetworkTypeMigrationInProgress" {
+				if cond.Status == "True" && cond.Reason == "NetworkTypeMigrationStarted" {
+					migrationInProgress = true
+					break
+				}
+
+				if cond.Status == "False" && cond.Reason == "NetworkTypeMigrationCompleted" {
+					fmt.Println("Network migration completed successfully!")
```

Comment:
Could you try to standardize so that all log messages are logged using GinkgoLogr? That way you can control verbosity through ginkgo and it writes the output to the correct spec report